src/main/java/uk/gov/hmcts/reform/sscs/ccd/service/UpdateCcdCaseService.java

```diff
@@ -43,23 +44,40 @@ public UpdateCcdCaseService(IdamService idamService,
     public SscsCaseDetails updateCaseV2(Long caseId, String eventType, String summary, String description, IdamTokens idamTokens, Consumer<SscsCaseDetails> mutator) {
         return updateCaseV2(caseId, eventType, idamTokens, caseDetails -> {
             mutator.accept(caseDetails);
-            return new UpdateResult(summary, description);
+            return new UpdateResult(caseDetails, summary, description);
         });
     }
 
+    @Retryable
+    public SscsCaseDetails updateCaseV2WithUnaryFunction(Long caseId, String eventType, String summary, String description, IdamTokens idamTokens, UnaryOperator<SscsCaseDetails> mutator) {
```

Comment:
What is UnaryOperator, can we use the existing Consumer function here?

Reply:
As the implementation is returning a new SscsCaseDetails so can't use the existing Consumer function.